### PR TITLE
Core/output frequency to output interval

### LINKED ISSUE #3914: [Output] output_frequency
The  time that passes between successive outputs is held in the json variable '**output_frequency**'. I find the name very unfortunate, since it clearly is not used as a frequency (units 1/time) but as a period (units of time). I suspect the difficulty of finding a suitable term guided this final choice. But I still think we could do better.
I do think it is much more intuitive to fix a time interval than a frequency (which would require calculations in most cases) or a number of time steps (one could fix the number of time steps in between prints).
So I propose to discuss possible alternatives (@KratosMultiphysics/implementation-committee ). Here are a few:
- output_period
- output_process_period
- output_time_interval
- time_between_outputs
- time_between_consecutive_outputs
- time_interval_between_consecutive_outputs
- time_interval_between_outputs
Or, my favourite so far:
- output_time_period

### PR DESCRIPTION
**Pull request by GuillermoCasas**
Replacing the variable name 'output_frequency' to 'output_interval' as agreed in #3914.

closes #3914

### CODE REVIEW

**Review by loumalouomega — COMMENTED** [review]
Should this be retrocomtabible and thrown a warning if the wrong one is considered?

**Comment by GuillermoCasas** [comment]
> Should this be retrocomtabible and thrown a warning if the wrong one is considered?

I think it should be as it is. A warning is thrown if the old name is used and the variable is then changed in the Parameters object. I have fixed all instances I could find of ["output_frequency"] in Kratos in my latest commit. The only really safe option I think is allowing the variable to be duplicated for some time, although this is undesirable I think...

**Comment by philbucher** [comment]
thanks @GuillermoCasas 
I will take a look and test it soon

**Comment by GuillermoCasas** [comment]
I have mainly changed core files, but also a few files in applications (4 or 5 in total) where there was going to be an obvious bug.

**Comment by ipouplana** [comment]
Good change! @jginternational could you take a look ?

**Comment by adityaghantasala** [comment]
Ping @philbucher  from @KratosMultiphysics/implementation-committee

**Comment by jginternational** [comment]
Implemented in https://github.com/KratosMultiphysics/GiDInterface/pull/731

Ready to merge when this merges

**Comment by philbucher** [comment]
I am slowly working my way through my TODO list, I will take a look at this in the upcoming days
Could you please merge master and solve the conflicts @GuillermoCasas ? Thx

**Comment by GuillermoCasas** [comment]
> I am slowly working my way through my TODO list, I will take a look at this in the upcoming days
> Could you please merge master and solve the conflicts @GuillermoCasas ? Thx

I will be doing it this week.

**Comment by AlejandroCornejo** [comment]
In behalf of the @KratosMultiphysics/implementation-committee, This PR is ready to merge whenever @philbucher agrees.

**Comment by GuillermoCasas** [comment]
ping @KratosMultiphysics/technical-committee.

**Comment by rubenzorrilla** [comment]
I think that Codacy suggestions might be errors. Can you please check?

**Comment by GuillermoCasas** [comment]
> I think that Codacy suggestions might be errors. Can you please check?

Done.

**Comment by philbucher** [comment]
@GuillermoCasas thanks for the patience!!!
This looks very good to me, but can you please move the copy-pasted deprecation code to a common location?
how about `kratos/python_scripts/deprecation_helpers.py`? Such a file could come in handy in the future

File: applications/SolidMechanicsApplication/python_scripts/restart_process.py
Comment (philbucher) [inline applications/SolidMechanicsApplication/python_scripts/restart_process.py:47]: this process is heavily used, please also make it backward compatible

I am surprised this even works in the tests, I thought I had tested it well

Comment (GuillermoCasas) [reply applications/SolidMechanicsApplication/python_scripts/restart_process.py:47]: I have applied the suggested changes

**Comment by GuillermoCasas** [comment]
> @GuillermoCasas thanks for the patience!!!
> This looks very good to me, but can you please move the copy-pasted deprecation code to a common location?
> how about `kratos/python_scripts/deprecation_helpers.py`? Such a file could come in handy in the future

I have added a similar 'deprecation_management.py'.

File: applications/SolidMechanicsApplication/python_scripts/restart_process.py
Comment (philbucher) [reply applications/SolidMechanicsApplication/python_scripts/restart_process.py:47]: sorry I confused this one with the core

**Comment by GuillermoCasas** [comment]
Please note that the codacy warnings have been addressed in #7246

File: kratos/python_scripts/tikz_output_process.py
Comment (philbucher) [inline kratos/python_scripts/tikz_output_process.py]:
```suggestion
```

same

File: kratos/python_scripts/unv_output_process.py
Comment (philbucher) [inline kratos/python_scripts/unv_output_process.py]:
```suggestion
```

I don't think the unv output has this setting, seems to be a copy paste mistake

File: kratos/python_scripts/deprecation_management.py
Comment (philbucher) [inline kratos/python_scripts/deprecation_management.py]:
```suggestion
    @staticmethod
    def HasDeprecatedVariable(context_string, settings, old_variable_name, new_variable_name):
```
this seems more appropriate here

File: kratos/python_scripts/deprecation_management.py
Comment (philbucher) [inline kratos/python_scripts/deprecation_management.py]:
```suggestion
    @staticmethod
    def ReplaceDeprecatedVariableName(settings, old_variable_name, new_variable_name):
```

File: kratos/python_scripts/deprecation_management.py
Comment (philbucher) [inline kratos/python_scripts/deprecation_management.py:3]: can you please add a bit of documentation?

**Review by philbucher — COMMENTED** [review]
It looks very good, cool concept!

I have a few minor comments after that I will approve, thx again!

File: applications/ParticleMechanicsApplication/python_scripts/particle_gid_output_process.py
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/particle_gid_output_process.py:76]: codacy has a point here, `context_string` is not defined!

File: kratos/python_scripts/unv_output_process.py
Comment (philbucher) [inline kratos/python_scripts/unv_output_process.py:17]: missing import:
`from  KratosMultiphysics.deprecation_management import DeprecationManager`

File: applications/DemStructuresCouplingApplication/python_scripts/sp_dem_fem_coupling_algorithm.py
Comment (philbucher) [inline applications/DemStructuresCouplingApplication/python_scripts/sp_dem_fem_coupling_algorithm.py:32]: missing import
`from  KratosMultiphysics.deprecation_management import DeprecationManager`

**Review by philbucher — COMMENTED** [review]
more comments based on codacy

File: kratos/python_scripts/deprecation_management.py
Comment (GuillermoCasas) [reply kratos/python_scripts/deprecation_management.py]: Ok, initially I was using the cls to pull out the string and I left it there...

File: kratos/python_scripts/unv_output_process.py
Comment (GuillermoCasas) [reply kratos/python_scripts/unv_output_process.py:16]: done

**Comment by GuillermoCasas** [comment]
@philbucher Thanks for the review. I have just uploaded the changes.

**Review by philbucher — APPROVED** [review]
perfect thx !!!

**Comment by GuillermoCasas** [comment]
I will add a test for the deprecation manager and extend its replacement type range when I get the chance

**Comment by rubenzorrilla** [comment]
The GUI needs to be updated accordingly.

**Comment by philbucher** [comment]
> The GUI needs to be updated accordingly.

https://github.com/KratosMultiphysics/GiDInterface/issues/723

**Pull request merged**
